resource.h doesn't compile with WINVER=0x0600

Technically, as per the "Update WINVER and _WIN32_WINNT" article from MSDN, targetting Windows Vista is an unsupported scenario, however in WIL headers it's implied to be the minimum supported version:

```cpp
// Minimal requirement set to Windows Vista
#if NTDDI_VERSION < NTDDI_VISTA 
#error WRL requires NTDDI_VERSION to be #defined at least to NTDDI_VISTA or greater
#endif
```

For `resource.h`, this is untrue however - attempting to include it and use it when targeting Vista throws a compilation error in `slim_event_t`, even if said feature is not used:
```
error C3861: 'QueryUnbiasedInterruptTime': identifier not found
```


Fix seems easy enough - this class is already guarded with the following preprocessor directive:
```
#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
```

and so extending it to exclude systems older than Windows 7 allows the code to compile:
```
#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && NTDDI_VERSION >= NTDDI_WIN7
```


EDIT:
This change is trivially PR-able, but does it require test coverage too? Windows 7 has test coverage, but since Vista is technically an unsupported target I would expect it not to require similar coverage and a best effort "support" might be sufficient.

--- a/include/wil/filesystem.h
+++ b/include/wil/filesystem.h
@@ -31,6 +31,7 @@ namespace wil
         return wcsncmp(path, L"\\\\?\\", 4) == 0;
     }
 
+#if (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
     //! Find the last segment of a path. Matches the behavior of shlwapi!PathFindFileNameW()
     //! note, does not support streams being specified like PathFindFileNameW(), is that a bug or a feature?
     inline PCWSTR find_last_path_segment(_In_ PCWSTR path)
@@ -51,6 +52,7 @@ namespace wil
         }
         return result;
     }
+#endif
 
     //! Determine if the file name is one of the special "." or ".." names.
     inline bool path_is_dot_or_dotdot(_In_ PCWSTR fileName)
@@ -83,7 +85,7 @@ namespace wil
         return false;
     }
 
-#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
     // PathCch.h APIs are only in desktop API for now.
 
@@ -957,7 +959,7 @@ namespace wil
         return result;
     }
 #endif // _CPPUNWIND
-#endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 }
 
 #endif // __WIL_FILESYSTEM_INCLUDED

--- a/include/wil/resource.h
+++ b/include/wil/resource.h
@@ -2676,7 +2676,7 @@ namespace wil
     typedef unique_any_t<event_t<details::unique_storage<details::handle_resource_policy>, err_exception_policy>>      unique_event;
 #endif
 
-#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
     enum class SlimEventType
     {
         AutoReset,
@@ -2843,7 +2843,7 @@ namespace wil
     /** An alias for `wil::slim_event_auto_reset`. */
     using slim_event = slim_event_auto_reset;
 
-#endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
     typedef unique_any<HANDLE, decltype(&details::ReleaseMutex), details::ReleaseMutex, details::pointer_access_none> mutex_release_scope_exit;
 

--- a/tests/common.h
+++ b/tests/common.h
@@ -345,7 +345,7 @@ namespace witest
         return S_OK;
     }
 
-#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
     struct TestFolder
     {

--- a/tests/wiTest.cpp
+++ b/tests/wiTest.cpp
@@ -3267,7 +3267,7 @@ TEST_CASE("WindowsInternalTests::ThreadPoolTimerTest", "[resource][unique_thread
     ThreadPoolTimerWorkHelper<wil::unique_threadpool_timer_nocancel, FILETIME>(SetThreadpoolTimer, true);
 }
 
-#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 static void __stdcall SlimEventTrollCallback(
     _Inout_ PTP_CALLBACK_INSTANCE /*instance*/,
     _Inout_opt_ void* context,
@@ -3339,7 +3339,7 @@ TEST_CASE("WindowsInternalTests::SlimEventTests", "[resource][slim_event]")
     }
 
 }
-#endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
 
 struct ConditionVariableCSCallbackContext
 {

